lib/src/sql/datetime.rs

```diff
@@ -168,19 +157,31 @@ fn nano(i: &str) -> IResult<&str, Datetime> {
 	let (i, nano) = nanosecond(i)?;
 	let (i, zone) = zone(i)?;
 
-	let v = match zone {
-		Some(z) => {
-			let d = z.ymd(year, mon, day).and_hms_nano(hour, min, sec, nano);
-			let d = d.with_timezone(&Utc);
-			Datetime(d)
-		}
-		None => {
-			let d = Utc.ymd(year, mon, day).and_hms_nano(hour, min, sec, nano);
-			Datetime(d)
-		}
-	};
+	convert(i, year, mon, day, hour, min, sec, nano, zone)
+}
 
-	Ok((i, v))
+fn convert(
+	i: &str,
+	year: i32,
+	mon: u32,
+	day: u32,
+	hour: u32,
+	min: u32,
+	sec: u32,
+	nano: u32,
+	zone: FixedOffset,
+) -> IResult<&str, Datetime> {
+	let n = NaiveDate::from_ymd_opt(year, mon, day)
+		.ok_or(Err::Error(error_position!(i, ErrorKind::Verify)))?;
+
+	let d = zone
+		.from_local_date(&n)
+		.unwrap()
+		.and_hms_nano_opt(hour, min, sec, nano)
+		.ok_or(Err::Error(error_position!(i, ErrorKind::Verify)))?
```

Comment:
Not sure if this is the correct way to send errors, tbh GitHub CoPilot wrote it. It works in my testing though.